csharp/ql/lib/ext/Microsoft.EntityFrameworkCore.model.yml

```diff
@@ -1,4 +1,17 @@
 extensions:
+  - addsTo:
+      pack: codeql/csharp-all
+      extensible: summaryModel
+    data:
+      - ["Microsoft.EntityFrameworkCore", "DbSet<>", False, "Add", "(T)", "", "Argument[0]", "Argument[this].Element", "value", "manual"]
+      - ["Microsoft.EntityFrameworkCore", "DbSet<>", False, "AddAsync", "(T)", "", "Argument[0]", "Argument[this].Element", "value", "manual"]
+      - ["Microsoft.EntityFrameworkCore", "DbSet<>", False, "AddRange", "(System.Collections.Generic.IEnumerable<T>)", "", "Argument[0].Element", "Argument[this].Element", "value", "manual"]
```

Comment:
Could you remind me about the semantics of the `WithElement`?
The other way of expressing that there flow from the elements of the input to the elements of the qualifier seems "intuitive" and easy to read.

Reply:
`WithElement` means that data must be stored inside a collection in order for flow to proceed. It may be a bit more performant, as we will not have to do first a read-step followed by an identical store-step, but instead just a simple local step (where we check that data is tracked inside an `Element`).

Reply:
Still strugling a bit with a mental model for this:
So is `A.Element -> B.Element` equivalent to `A.WithElement -> B` or is there a difference?

Reply:
Yes, those are equivalent, but the latter could in principle mean better performance when computing global data flow.

Reply:
Alright that makes sense, but then it is purely an implementation detail.
One more sanity check of my understanding:
- `A -> B` is equivalent to `A.WithoutElement -> B` & `A.WithElement -> B`.

Reply:
Also, is it worth it that we make an issue to convert all `A.Element -> B.Element` like summaries (or maybe even better apply the above equivalence in the implementation).

Reply:
We could consider making this in the implementation, but it would have to be C# specific.